src/Sarif/FileRegionsCache.cs

```diff
@@ -184,17 +186,15 @@ public Region ConstructMultilineContextSnippet(Region inputRegion, Uri uri, stri
             region.EndColumn = 0;
             region.StartLine = 0;
             region.EndLine = 0;
-            region.CharOffset = originalRegion.CharOffset < smallSnippetLength
-                ? 0
-                : originalRegion.CharOffset - smallSnippetLength;
+            region.CharOffset = Math.Max(0, originalRegion.CharOffset - SMALLSNIPPETLENGTH);
```

Comment:
>CharOffset [](http://example.com/codeflow?start=19&length=10)

Any prepended context region must start at the beginning of the file, any value that precedes this is invalid.